internal/graph/dispatch_throttling_check_resolver_test.go

```diff
@@ -2,231 +2,174 @@ package graph
 
 import (
 	"context"
-	"sync"
 	"testing"
-	"time"
 
 	"github.com/stretchr/testify/require"
 	"go.uber.org/goleak"
 	"go.uber.org/mock/gomock"
 
-	"github.com/openfga/openfga/pkg/telemetry"
+	mocks "github.com/openfga/openfga/internal/mocks"
+	"github.com/openfga/openfga/internal/throttler/threshold"
 )
 
 func TestDispatchThrottlingCheckResolver(t *testing.T) {
 	t.Cleanup(func() {
 		goleak.VerifyNone(t)
 	})
-	t.Run("dispatch_below_threshold_resolves_immediately", func(t *testing.T) {
+	t.Run("dispatch_below_threshold_doesnt_call_throttle", func(t *testing.T) {
 		ctrl := gomock.NewController(t)
 		defer ctrl.Finish()
 
 		dispatchThrottlingCheckResolverConfig := DispatchThrottlingCheckResolverConfig{
-			// We set timer ticker to 1 hour to avoid it interfering with test
-			Frequency:        1 * time.Hour,
 			DefaultThreshold: 200,
 			MaxThreshold:     200,
 		}
-		dut := NewDispatchThrottlingCheckResolver(dispatchThrottlingCheckResolverConfig)
-		defer dut.Close()
+		mockThrottler := mocks.NewMockThrottler(ctrl)
 
-		initialMockResolver := NewMockCheckResolver(ctrl)
-		dut.SetDelegate(initialMockResolver)
+		dut := NewDispatchThrottlingCheckResolver(
+			WithDispatchThrottlingCheckResolverConfig(dispatchThrottlingCheckResolverConfig),
+			WithThrottler(mockThrottler),
+		)
```

Comment:
Please `t.Cleanup(dut.Close)` here and elsewhere in this file. Resources should be closed properly.